magenta-lib/src/main/scala/magenta/tasks/tasks.scala

```diff
@@ -147,18 +147,14 @@ case class S3Upload( stack: Stack,
     s"$stackName$stageName${toRelative(file)}"
   }
 
-  def contentTypeLookup(fileName:String) = {
-    import org.parboiled2._
-    import org.parboiled2.CharPredicate._
-    class ExtensionParser(val input: ParserInput) extends Parser {
-      def FileName = rule { oneOrMore(Visible -- '.') ~ '.' ~ Extension ~ EOI }
-      def Extension = rule { capture(oneOrMore(Visible)) ~> (s => mimeTypeMap.get(s)) }
-    }
-    new ExtensionParser(fileName).FileName.run().getOrElse(None)
-  }
+  def contentTypeLookup(fileName: String) =
+    fileExtension(fileName).flatMap(mimeTypeMap.get)
 
   def cacheControlLookup(fileName:String) = cacheControlPatterns.find(_.regex.findFirstMatchIn(fileName).isDefined).map(_.value)
 
+  private def fileExtension(fileName: String) =
+    fileName.split('.').drop(1).lastOption
```

Comment:
Ah, that makes sense. I forgot that `split` included the original if it didn't find the character.